### ISSUE TITLE
Run a job on defined endTime

### ISSUE THREAD

**Issue by tedbrun** [yahoo/sherlock#35]
Hello folks,

I am trying to run a job with Sherlock on **event time data** that implies that at run time I have not all the data. I therefore would like to run a daily job, which query druid until two days before the run time. But it does not seem to work. When I query druid by setting in my query `"intervals": "ZonedDateTime.now().minusDays(18),ZonedDateTime.now().minusDays(2)"` it seems that Sherlock still queries on the run time.

In Flash query, we have the possibility to change the end Time of the query in the UI. Something strange to me is that when I set in the query Druid: `"intervals": "ZonedDateTime.now().minusDays(18),ZonedDateTime.now().minusDays(2)"` and I change in the UI the end Date of the query and set a day different from `ZonedDateTime.now().minusDays(2)` , Sherlock will consider the **endDate defined in the UI**. Does that mean that Sherlock does not take into account the values set in the intervals of query Druid?

In the setting of the jobs, it is not possible to set the End Date (which is normal because this date will change every day) as in the Flash Query. But it seems that the end Date run each day for the job is actually Date.now() even if in my intervals of the query Druid, I have set `ZonedDateTime.now().minusDays(2)`  which is what I need. Have you got any idea how I could run a job that returns an anomaly detection 2 days before the current date?

Thanks all for your help

**Comment by jigs1993** [comment]
Hi @tedbrun 

As per I understand from the above, let say you are running a job and the job run time is `t` in the queue. So at time `t` when the job runs sherlock  would build the query interval as per this formula

> [getEndTimeForInterval](https://github.com/yahoo/sherlock/blob/master/src/main/java/com/yahoo/sherlock/enums/Granularity.java#L176)([`t` - druid cluster SLA](https://github.com/yahoo/sherlock/blob/master/src/main/java/com/yahoo/sherlock/service/SchedulerService.java#L347))

**for example:** if your druid cluster SLA is 2 days and job run time is 28th Oct 4:32pm then according to above formula your **query interval endtime** in druid query  would be: 

getEndTimeForInterval(`28th Oct 4:32pm` minus `2days`) = getEndTimeForInterval(`26th Oct 4:32pm`) = **26th Oct 00:00** 

basically `getEndTimeForInterval()` function does _Floor operation_ on given time according to the _granularity_ of your query

to simply answer to your question, i would suggest you to add druid cluster SLA to be 48 hours

**Comment by tedbrun** [comment]
Hello @jigs1993 it works. I really thank you.
